pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -68,6 +68,37 @@ const (
 )
 
 const (
+	activateRawOsdOnNodeScript = `
+set -o errexit
+set -o pipefail
+set -o nounset # fail if variables are unset
+set -o xtrace
+
+OSD_ID="$ROOK_OSD_ID"
+
+OSD_LIST="$(mktemp)"
+
+# 'ceph-volume raw list' (which the osd-prepare job uses to report OSDs on nodes)
+#  returns user-friendly device names which can change when systems reboot. To
+# keep OSD pods from crashing repeatedly after a reboot, we need to scan the
+# disks in an init container to find the correct one for the desired OSD ID.
+ceph-volume raw list > "$OSD_LIST"
```

Comment:
This always outputs json?

Reply:
Yep.

Reply:
Why not list against `$ROOK_BLOCK_PATH` first and if there is a mismatch, go through an entire listing? This will save processing time.

Reply:
It will save 100-150ms. I'm not sure that is worth the optimization.

Reply:
On 36 disks machines with a loaded system, it might make a difference though :)